pkg/datapath/linux/node.go

```diff
@@ -680,15 +677,18 @@ func (n *linuxNodeHandler) insertNeighbor(ctx context.Context, newNode *nodeType
 		logfields.Interface: n.neighDiscoveryLink.Attrs().Name,
```

Comment:
I believe you have a race here. `n.neighDiscoveryLink` may be updated concurrently:
https://github.com/cilium/cilium/blob/1309d2aa936f50b73ceb7e47458260ae9d7e8b15/pkg/datapath/linux/node.go#L1364

Setting a pointer in one go routine and reading it in another without synchronization is a data race I believe. Before this PR, this code here safe because `insertNeighbor` held the `n.mutex` lock. 

You also have more unprotected accesses to `n.neighDiscoveryLink` below:
https://github.com/cilium/cilium/blob/1309d2aa936f50b73ceb7e47458260ae9d7e8b15/pkg/datapath/linux/node.go#L743

Reply:
Good catch! Probably I should use `sync/atomic.{Load,Swap}Pointer()` for n.neighDiscoveryLink, and return early in the `insertNeighbor()` if its nil.

Reply:
Actually it is enough just to check whether `neighDiscoveryLink` is set in `insertNeighbor()`.